src/Features/LanguageServer/Protocol/Handler/DocumentChanges/DidOpenHandler.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.DocumentChanges
+{
+    [Shared]
+    [ExportLspMethod(LSP.Methods.TextDocumentDidOpenName, mutatesSolutionState: true)]
+    internal class DidOpenHandler : IRequestHandler<LSP.DidOpenTextDocumentParams, object>
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public DidOpenHandler()
+        {
+        }
+
+        public LSP.TextDocumentIdentifier? GetTextDocumentIdentifier(LSP.DidOpenTextDocumentParams request)
+            => new LSP.TextDocumentIdentifier { Uri = request.TextDocument.Uri };
```

Comment:
So in the case of linked files, which document is this? The Uri isn't sufficient to distinguish, right?

Reply:
I don't know, but will check. For interest, this is where the document identifier becomes a document: https://github.com/dotnet/roslyn/blob/master/src/Features/LanguageServer/Protocol/Extensions/Extensions.cs#L109